ddl/ddl_worker.go

```diff
@@ -367,14 +370,24 @@ func (d *ddl) addBatchDDLJobs2Table(tasks []*limitJobTask) error {
 		return nil
 	})
 	if err == nil {
-		jobTasks := make([]*model.Job, len(tasks))
+		jobTasks := make([]*model.Job, 0, len(tasks))
```

Comment:
Need to take care about the memory usage here. There could be a lot of job currently running.

Reply:
This should be the same logic as before, preallocate the same amount of memory, except that the final length of the array is uncertain.